scripts/bb_sim.py

```diff
@@ -98,10 +113,16 @@ def cb_amp_lf(*x, **y):
     # min_vs / (5.0 * hh)
     args.flo = 0.5 / (5.0 * lf.hh)
 
+if is_master:
+    # Logging each argument
+    for key in vars(args):
+        logger.debug("{} : {}".format(key, getattr(args, key)))
+
+comm.Barrier() # prevent other processes from messing log file until master is done with logging above
```

Comment:
not required if you write the message at once.
I don't think it's a good idea to loop and send lines individually to a logger as it defeats the purpose.
logger.debug("\n".join(["   " for key in var(args)]))
no more Barrier.